### DISCUSSION TITLE
Bölüşülen pastada en çok payı kim alır sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/BilsemPython#50] [Soru Cevap] [answered]
100 kişi bir pastayı bölüşmektedir. 1. kişi pastanın %1 ini, 2. kişi kalan pastanın %2 sini, 3. kişi kalan pastanın %3 nü ve böylece devam edip 100. kişi de kalan pastanın %100 ünü alır. Soru en çok payı kaçıncı kişi almıştır.

(Önce algoritmasını kendi cümlelerinizle yazınız daha sonra kodunuzu ekleyiniz, kodunuzu kod bloğuna almayı unutmayınız)

**Comment by Furkanyildirm** [comment]
```
pay=[]
b=100
for i in range(1,101):
    pay.append(b*(0.01*i))
    b=b-b*(0.01*i)
a=max(pay)
print("en çok yiyen pastanın",a,"kadarını yemiş")

#payları içinde tutacak "pay"ı oluştur,
#pastanın 100'desini bir değişkene ata,
#100 kişiyi teker teker yoklayayacak bir döngü oluştur,
#yenilen payı pay'a ekle,
#pasta yüzdesinden yenilen kısmı çıkar,
#payın içindeki değerlerin en büyüğünü bulmak için max kullan,
#en büyüğü ekrana bastır ekrana bastır.
```

**Comment by CinarSatir** [comment]
```python
#degisken verdim
d = []
b = 100
c = 0.01
#kisi sayısını i yaptım a sayısı yenilen miktar yenilen miktarı d dizesine atıp sıraladım kucukten buyuye sonrada en sonuncu elemanı aldım
for i in range(1,101):
    a = b*c
    d.append(a)
    b = b-a
    c = c+0.01
    print(a)
d.sort()
print("En fazla pasta alan kisi pastanın yüzde",d[-1],"kısmınıalmıstır")
```

**Comment by slos8888** [comment]
```
pay=[]
b=100
for i in range(1,101):
    pay.append(b*(0.01*i))
    b=b-b*(0.01*i)
a=max(pay)
print("en çok yiyen pastanın",a,"kadarını yemiş")

```

**Reply by Furkanyildirm** [reply]
bir şey soracağım kopyala yapıştır yapmadığına eminsin dimi çünkü benimkiyle tıpa tıp aynı
benimki:
pay=[]
b=100
for i in range(1,101):
    pay.append(b*(0.01*i))
    b=b-b*(0.01*i)
a=max(pay)
print("en çok yiyen pastanın",a,"kadarını yemiş")

**Comment by OmerAkbayrak** [comment]
```python
a=[]#Sıralama için Liste
kisi=0#Kişi Sayısı
b=100#Pastanın Bütünü
s=0#Yüzdeler değişkeni


for i in range (1,101): #100 kere çalışacak döngü
    kisi+=1
    
    s+=0.01 #Yüzde bulma 0.01 artıyor
    oran = b*s #yenen pasta sayısı
    a.append(oran) #Listeye yeneni ekler
    b-=oran #bütünden yenilen çıkarılır
    
    if max(a)>oran: #Eğer Oran en büyükten küçükse döngüden çıkar
        break
    

print("En çok",kisi,max(a),"Dilim Pasta yenmiş.")#Sonuçları Yazdırır

    
    
    

```

**Comment by OmerB2** [comment]
Pastanın yüzdesi belirtilir,
en büyük oran belirtilir(o an için),
100 kişinin hepsine bakacak bir döngü oluşturulur,
pay değişkeni ile her bir kişinin pastanın yüzde kaçını alacağı belirtilir,
pay eb değişkeninden büyükse eb değişkeni paya eşitlenir,
ec değişkeni kişi değişkenine eşitlenir,
pastadan alınan yüzde çıkarılır,
döngünün sonuna gelindiğinde en çok payı kimin aldığı yazdırılır.
```
p=100
eb=0
for k in range(1,101):
    pay=p*(k/100)
    if pay>eb:
        eb=pay
        ec=k
    p=p-pay
print(ec,". kişi en çok payı alır.")
```

**Comment by Sadik1Dev** [comment]
#önce pasta değişkenine pasta miktarı kolaylık olsun diye 1000 olarak tanımlarım
#döngü içinde en büyük pay ve kişinin sırasını bulmak için değişkenler tanımladım
#1'den 100 e kadar bir döngü içinde pastanın sıra sayısı ile yüzdelik olarak paylaşımı yapılır
#paylaşılan pasta mevcut ilk pasta miktarından çıkılır
#paylaşım miktari enbuyukpay diye bir değişkene atanıp her paylaşımda bu değişken kontrol edilip bundan büyük değer atanır
#döngü sonunda enbuyuk pay ve sirası yazdırılır.

```
pasta = 1000
sira = 0
enbuyuk = 0
for x in range(1,101):
    y=pasta*(x/100)
    if y > enbuyuk:
        enbuyuk = y
        sira = x
    pasta = pasta-y



print("en büyük payı alan kişi",sira,". kişi aldığı pay:",enbuyuk)

```

**Comment by OmerMete** [comment]
İlk başta değişkenleri ayarladık. For un içinde yüzdeyi buluyoruz listenin içine atıyoruz sort yapıyoruz ve en büyüğünü listenin içinde -1 i alıp yazdırıyoruz.
```
a=[]
b=100
c=0.01
for x in range (1,101):
  d =b*c
  a.append(d)
  b=b-d
  c=c+0.01
a.sort()
print("pastadan en çok alan kişi pastanın yüzde",a[-1],"kısmını yemiştir")
```

**Comment by AdaNilPoyraz** [comment, marked as answer]
İlk olarak değişkenleri oluşturalım
```python
pasta = 100
enb = 0
kisi = 0
```
Bir for döngüsü oluşturarak herkese düşen payı bulalım
```python
for k in range(1,101):
    pay = pasta * (k/100)
```
Şimdi bu pay en büyük pay mı onu kontrol edip pastayı küçültmemiz lazım
```python
    if pay > enb:
        enb = pay
        kisi = k
    pasta -= pay
```
Son olarak da kişi ve pastadan aldığı payı yazdırıyoruz
```python
print(kisi," = ",enb)
```

Hepsi beraber;
```python
pasta = 100
enb = 0
kisi = 0
for k in range(1,101):
    pay = pasta * (k/100)
    if pay > enb:
        enb = pay
        kisi = k
    pasta -= pay
print(kisi," = ",enb)

```

**Comment by AcelyaP** [comment]
İlk olarak değişkenlerimizi belirliyoruz.
Sonra bir tane for döngüsü kuruyor ve herkese düşen payı buluyoruz.
Aldığımız pay en büyük pay mı diye kontrol ediyor ve pastayı küçültüyoruz.
İşimiz bittiğinde kişiyi yazdırıyoruz.
```
pasta=100

ebüy=0

for i in range(1,101):

    pay=pasta*(i/100)

    if pay>ebüy:

        ebüy=pay

        kişi=i

    pasta=pasta-pay

print(kişi,".kişi en çok pasta payını alır.")
```

**Comment by Ardemin** [comment]
Arın Demir İnan
```
pasta=100
eb=0
ebk=0
for i in range(1,100):
    pay=pasta*(i/100)
    if pay > eb:
        eb=pay
        ebk=i
    pasta=pasta-pay
print("kisi sayisi",ebk,"aldığı miktar", eb)

```
